### PR TITLE
fix 10685: test suite related bug fixes and improvements

### LINKED ISSUE #10685: UI: Test suite related improvements
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**

- [ ] Deleting a test suite redirects to the homepage. It should redirect to the test suites list page
- [ ] Provide deletion option for test suites inside table rows.
- [ ] Unnecessary fields are being sent to the API call for fetching the test results in the entity preview panel.

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes:

Fixes #10685

I worked on test suite related bug fixes and improvements:

- [x] Redirection to homepage issue after test suite deletion

https://user-images.githubusercontent.com/51777795/229542220-70cd72fb-6d4a-4f87-bfdd-7d2292654f8d.mov

- [x] Removed unnecessary fields being passed to the API to fetch the test case results for the entity summary panel as well as the dataset details page
- [x] Added Action column in test suite listing table for deletion of test suites from the table

https://user-images.githubusercontent.com/51777795/229542769-d0f3692b-1bd3-4020-8dce-355aa69361ae.mov

- [x] Fixed issue with the test case creation form where going back and forth through the steps was making the fields unresponsive.

https://user-images.githubusercontent.com/51777795/229542940-4d4c723a-3872-4112-a85c-9a578c68c6c5.mov

#
### Type of change:

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx [hunk 1]
```diff
@@ -151,6 +167,37 @@ const TestSuitePage = () => {
...
+                  icon={<IconDelete width={16} />}
+                  type="text"
+                  onClick={(e) => {
+                    // preventing expand/collapse on click of delete button
+                    e.stopPropagation();
+                    setSelectedTestSuite(record);
+                  }}
+                />
+              </Tooltip>
+            </Row>
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx:197]: row must have col as child

File: openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx [hunk 2]
```diff
@@ -269,6 +316,18 @@ const TestSuitePage = () => {
...
+          allowSoftDelete={!showDeleted}
+          entityId={selectedTestSuite?.id || ''}
+          entityName={selectedTestSuite?.name || ''}
+          entityType="testSuite"
+          prepareType={false}
+          visible={!isUndefined(selectedTestSuite)}
+          onCancel={() => {
+            setSelectedTestSuite(undefined);
+          }}
+        />
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx:330]: can you try this with the latest main? 
- merge main in your PR
- re-build
- and try this action
API will fail, we need to remove `prepareType={false}`, after that API will pass
ref- https://github.com/open-metadata/OpenMetadata/issues/10041

**Review by ShaileshParmar11 — CHANGES_REQUESTED** [review]
left few comments

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/TableProfilerV1.tsx [hunk 3]
```diff
@@ -250,7 +250,7 @@ const TableProfilerV1: FC<TableProfilerProps> = ({
     setIsTestCaseLoading(true);
     try {
       const { data } = await getListTestCase({
-        fields: 'testCaseResult,entityLink,testDefinition,testSuite',
```
[the remark this answers is not in the document]
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/TableProfilerV1.tsx:253]: No, we are only taking testCaseResults from the response

File: openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/TestSuitePage/TestSuitePage.tsx:330]: Ok I'll try that

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**